### ISSUE TITLE
[REC] Firemage using user created profile does not show up unless combustion has been used

### ISSUE THREAD

**Issue by nhawrot** [Hekili/hekili#2806]
### Before You Begin

- [X] I confirm that I have downloaded the latest version of the addon.
- [X] I am not playing on a private server.
- [X] I checked for an [existing, open ticket](https://github.com/Hekili/hekili/labels/recommendation) for this issue and was not able to find one.
- [X] I edited the title of this issue (above) so that it describes the issue I am reporting.
- [ ] I am reporting an issue with the default priority included with the specialization (imported or edited priorities are not supported).

### Describe the Issue

I've been using Hekili for a long time and love it.  I understand there are significant limitations when used with any mage spec, but honestly, I don't mind too much.  I've set up a profile that I use for all of my characters with a layout that is conducive to my playstyle.  Since this last Tuesday, the Hekili window does not show for my mage unless combustion has been triggered.  It works perfectly fine for my other characters, and it works using the default/alternate profiles, just not the one I created.  I don't know why this is.  Any help would be great.  I'll include screenshots and the snapshots below to help.

I can duplicate this issues regularly.  If combustion is off cooldown, and I don't have the combustion buff, Hekili will not show.  If combustion is on cooldown, or I have the buff, it will show my Hekili layout.  Hekili will stay on the screen permanently after using combustion, even after the buff is gone and the spell comes off cooldown.  However, if I do a /reload, or restart the game, it disappears again.

I've tried verifying that most/all of my settings match the default, except the layout which could be causing problems.

Pastebin 1 after clean reload with no combustion, and Hekili not showing:  https://pastebin.com/gC3BqUCS
Pastebin 2 after clean reload, and using combustion, and Hekili is showing:  https://pastebin.com/e7WG2kWx
Pastebin 3 same as above, additional items in drop down:  https://pastebin.com/A8jRNFk2, https://pastebin.com/aPwqcaG8, https://pastebin.com/yQqkdSDW

Here is the pastebin when swapping to the default layout, after a clean reload and with no combustion and Hekili is showing:  https://pastebin.com/gYauvUUR

![1](https://github.com/Hekili/hekili/assets/145692930/8b3b9d2b-3881-4427-8048-2060f0c799b2)
![2](https://github.com/Hekili/hekili/assets/145692930/e66c578b-0639-46fc-9aeb-e0cbd82ad54d)
![3](https://github.com/Hekili/hekili/assets/145692930/8d1f3a4e-5fe8-4e66-ac2f-1fc6f0a00ae5)

**Screenshots:**
1 = fresh reload
2 = doing damage to target dummy and Hekili recommendations don't show
3 = combustion used and suddenly it shows up.

**Character Data:**
build: v10.1.17-1.0.1
level: 70 (70)
class: MAGE
spec: fire

talents: B8DArSxcnei16P8xFL3rzzOyRKRIFNkEJpBikEREpIJAAAAAAAAARSSkIRJRRSikkkkGAAAIA
    alexstraszas_fury = 1/1
    alter_time = 1/1
    arcane_warding = 2/2
    blast_wave = 1/1
    blazing_barrier = 1/1
    call_of_the_sun_king = 1/1
    combustion = 1/1
    conflagration = 1/1
    critical_mass = 2/2
    dragons_breath = 1/1
    energized_barriers = 1/1
    feel_the_burn = 1/1
    fervent_flickering = 1/1
    fevered_incantation = 1/1
    fiery_rush = 1/1
    fire_blast = 1/1
    firemind = 1/1
    flame_on = 2/2
    flame_patch = 1/1
    freezing_cold = 1/1
    frigid_winds = 2/2
    fuel_the_fire = 1/1
    greater_invisibility = 1/1
    ice_block = 1/1
    ice_cold = 1/1
    ice_floes = 1/1
    improved_combustion = 1/1
    incanters_flow = 1/1
    incendiary_eruptions = 1/1
    kindling = 1/1
    mass_barrier = 1/1
    master_of_time = 1/1
    meteor = 1/1
    mirror_image = 1/1
    overflowing_energy = 1/1
    phoenix_flames = 1/1
    phoenix_reborn = 1/1
    pyroblast = 1/1
    pyrotechnics = 1/1
    quick_witted = 1/1
    reduplication = 1/1
    remove_curse = 1/1
    scorch = 1/1
    searing_touch = 1/1
    shifting_power = 1/1
    spellsteal = 1/1
    sun_kings_blessing = 1/1
    surging_blaze = 1/1
    tempest_barrier = 2/2
    temporal_velocity = 2/2
    temporal_warp = 1/1
    time_manipulation = 1/1
    tome_of_antonidas = 1/1
    tome_of_rhonin = 1/1
    wildfire = 2/2

pvptalents: none

covenant: none

conduits: none

soulbinds: []

sets: irideus_fragment = 1
    screaming_black_dragonscale = 1
    tier30 = 4

gear: crackling_codex_of_the_isles = 1
    discarded_creations_restraint = 1
    elemental_lariat = 1
    irideus_fragment = 1
    obsidian_seared_hexsword = 1
    onyx_impostors_birthright = 1
    screaming_black_dragonscale = 1
    shaldorei_tabard = 1
    signet_of_titanic_insight = 1
    tier30 = 4
    underlight_conjurers_arcanocowl = 1
    underlight_conjurers_aurora = 1
    underlight_conjurers_bracelets = 1
    underlight_conjurers_gloves = 1
    underlight_conjurers_trousers = 1
    vibrant_wildercloth_slippers = 1
    vibrant_wildercloth_vestments = 1
    voice_of_the_silent_star = 1

legendaries: none

itemIDs: 157759, 190511, 192999, 193001, 193509, 193519, 193743, 194879, 202547, 202549, 202550, 202551, 202552, 202612, 202652, 204398, 204465

settings: aoe = 3
    combatRefresh = 0.25
    custom1Name = Custom 1
    custom2Name = Custom 2
    cycle = false
    cycle_min = 6
    damage = true
    damageDots = false
    damageExpiration = 6
    damageOnScreen = true
    damagePets = false
    damageRange = 0
    enabled = true
    enhancedRecheck = true
    gcdSync = false
    maxTime = 20
    nameplateRange = 8
    nameplates = false
    noFeignedCooldown = true
    package = Fire
    petbased = false
    potion = spectral_intellect
    potionsReset = 20180919.1
    regularRefresh = 0.5
    throttleRefresh = false
    throttleTime = false
    check_explosion_range = true
    prevent_hardcasts = true
    pyroblast_pull = true

toggles: cooldowns = true [separate]
    custom1 = false 
    custom2 = false 
    defensives = true [separate]
    essences = true [overridden]
    interrupts = true [separate]
    mode = automatic 
    potions = false 

keybinds: 
    alter_time                  = SF  [03]
    arcane_explosion            = X   [01]
    arcane_intellect            = C6  [08]
    arcane_pulse                = C   [01]
    blast_wave                  = S4  [03]
    blazing_barrier             = AZ  [02]
    blink                       = 4   [01]
    combustion                  = 2   [01]
    cone_of_cold                = AM3 [02]
    counterspell                = M3  [01]
    dragons_breath              = SM3 [03]
    fire_blast                  = E   [01]
    fireball                    = AQ  [02]
    flamestrike                 = SE  [03]
    frost_nova                  = A4  [02]
    frostbolt                   = A3  [02]
    greater_invisibility        = SZ  [03]
    healthstone                 = SR  [03]
    ice_cold                    = S1  [03]
    ice_floes                   = AF  [02]
    ice_nova                    = S5  [08]
    irideus_fragment            = SX  [03]
    living_bomb                 = 3   [01]
    mass_barrier                = SC  [03]
    meteor                      = A2  [02]
    mirror_image                = SQ  [03]
    phoenix_flames              = 1   [01]
    polymorph                   = F   [01]
    pyroblast                   = AE  [02]
    remove_curse                = Q   [01]
    scorch                      = A1  [02]
    screaming_black_dragonscale = AX  [02]
    shifting_power              = S3  [03]
    spellsteal                  = AC  [02]
    time_warp                   = S2  [03]

warnings: [#1] Update: Interface/AddOns/Hekili/Dragonflight/MageFire.lua:940: attempt to compare number with boolean
[string "@Interface/AddOns/Hekili/Dragonflight/MageFire.lua"]:940: in function <Interface/AddOns/Hekili/Dragonflight/MageFire.lua:939>
[string "=(tail call)"]: ?
[string "-- Fire:default:20
return safebool(  not (safebool(variable.item_cutoff_active))  )"]:2: in main chunk
[string "=(tail call)"]: ?
[string "@Interface/AddOns/Hekili/Core.lua"]:845: in function `GetPredictionFromAPL'
[string "@Interface/AddOns/Hekili/Core.lua"]:1397: in function `GetNextPrediction'
[string "@Interface/AddOns/Hekili/Core.lua"]:1777: in function <Interface/AddOns/Hekili/Core.lua:1446>

[#2] Update: Interface/AddOns/Hekili/Dragonflight/MageFire.lua:940: attempt to compare number with boolean
[string "@Interface/AddOns/Hekili/Dragonflight/MageFire.lua"]:940: in function <Interface/AddOns/Hekili/Dragonflight/MageFire.lua:939>
[string "=(tail call)"]: ?
[string "-- Fire:default:20
return safebool(  not (safebool(variable.item_cutoff_active))  )"]:2: in main chunk
[string "=(tail call)"]: ?
[string "@Interface/AddOns/Hekili/Core.lua"]:845: in function `GetPredictionFromAPL'
[string "@Interface/AddOns/Hekili/Core.lua"]:1397: in function `GetNextPrediction'
[string "@Interface/AddOns/Hekili/Core.lua"]:1683: in function <Interface/AddOns/Hekili/Core.lua:1446>

### How to Reproduce

1) Enter game as firemage
2) Go to training dummy
3) Use my created profile
4)  If combustion is off cooldown, and I don't have the combustion buff, Hekili will not show.  
5)  If combustion is on cooldown, or I have the buff, it will show my Hekili layout.  Hekili will stay on the screen permanently after using combustion, even after the buff is gone and the spell comes off cooldown.  However, if I do a /reload, or restart the game, it disappears again.
6) do /reload in WoW, then use default profile.  It shows no problem.  

### Snapshot (Link)

https://pastebin.com/gC3BqUCS
https://pastebin.com/e7WG2kWx
https://pastebin.com/A8jRNFk2
https://pastebin.com/aPwqcaG8
https://pastebin.com/yQqkdSDW
https://pastebin.com/gYauvUUR

### Raidbots Sim Report (Link)

https://www.raidbots.com/simbot/report/o8ZFx8169Ft1Ma9KsngoDU

### Contact Information

I can be reached on discord at n8dawg1

**Comment by Hekili** [comment]
If you're comfortable entering code, you can try editing Dragonflight/MageFire.lua and adding ` == 0` after `cooldown.item_cd_1141.remains` like this:

```Lua
-- actions+=/variable,name=item_cutoff_active,value=(variable.time_to_combustion<variable.on_use_cutoff|buff.combustion.remains>variable.skb_duration&!cooldown.item_cd_1141.remains)&((trinket.1.has_cooldown&trinket.1.cooldown.remains<variable.on_use_cutoff)+(trinket.2.has_cooldown&trinket.2.cooldown.remains<variable.on_use_cutoff)>1)
spec:RegisterVariable( "item_cutoff_active", function ()
    return ( variable.time_to_combustion < variable.on_use_cutoff or buff.combustion.remains > variable.skb_duration and cooldown.item_cd_1141.remains == 0 ) and ( ( trinket.t1.has_use_buff and trinket.t1.cooldown.remains < variable.on_use_cutoff ) and ( trinket.t2.has_use_buff and trinket.t2.cooldown.remains < variable.on_use_cutoff ) > 1 )
end )
```

That *should* address the error noted in your snapshots.  I'll post that fix later today; let me know if you try it ahead of time and still have issues.

**Comment by nhawrot** [comment]
I went ahead and made these changes, however when I do this now the hekili doesn't show up in any _firemage_ profiles.  Seems to have broken it, unless I copied it wrong?  I think I did it correctly though.  This is the .txt file, not the .lua file.

[MageFire.txt](https://github.com/Hekili/hekili/files/12686791/MageFire.txt)

**Comment by Hekili** [comment]
I'm not sure what you mean, but I'd guess that there is some kind of error message either in /hekili > Warnings or captured by BugSack and BugGrabber.  You can just revert and wait for me to post my fix, and then follow up.

**Comment by nhawrot** [comment]
Sorry, I should clarify.  Now in Hekili the firemage profiles don't show up.  See image.
![4](https://github.com/Hekili/hekili/assets/145692930/d903fee4-1a10-4a02-9249-89c6ad72e1cb)

**Comment by Hekili** [comment]
I'd guess that there is some kind of error message either in /hekili > Warnings or captured by BugSack and BugGrabber.

**Comment by Hekili** [comment]
Oh, it looks like there are stray marks in the example I provided you.  You want to delete the **`** characters.  I updated the code block above.

**Comment by nhawrot** [comment]
This fixed the problem of not showing the "Fire" option in priorities, but the initial problem still remains, although it is now slightly different.  When I pop combustion, the hekili display doesn't show up (like it was before) and now it seems like there is a ~5 second delay before the display shows up.  If I start with a fresh reload, combustion off CD and no combustion buff I see no Hekili (for my own user created profile).  Then, I pop combustion and wait about 5 seconds and then the display shows up.

1.

Update: Interface/AddOns/Hekili/Dragonflight/MageFire.lua:940: attempt to compare number with boolean

[string "@Interface/AddOns/Hekili/Dragonflight/MageFire.lua"]:940: in function <Interface/AddOns/Hekili/Dragonflight/MageFire.lua:939>
    [string "=(tail call)"]: ?
    [string "-- Fire:default:20
return safebool(  not (safebool(variable.item_cutoff_active))  )"]:2: in main chunk
    [string "=(tail call)"]: ?
    [string "@Interface/AddOns/Hekili/Core.lua"]:845: in function `GetPredictionFromAPL'
[string "@Interface/AddOns/Hekili/Core.lua"]:1397: in function `GetNextPrediction'
        [string "@Interface/AddOns/Hekili/Core.lua"]:1777: in function <Interface/AddOns/Hekili/Core.lua:1446>
      
2.      
Update: Interface/AddOns/Hekili/Dragonflight/MageFire.lua:940: attempt to compare number with boolean

[string "@Interface/AddOns/Hekili/Dragonflight/MageFire.lua"]:940: in function <Interface/AddOns/Hekili/Dragonflight/MageFire.lua:939>
[string "=(tail call)"]: ?
[string "-- Fire:default:20
return safebool(  not (safebool(variable.item_cutoff_active))  )"]:2: in main chunk
[string "=(tail call)"]: ?
[string "@Interface/AddOns/Hekili/Core.lua"]:845: in function `GetPredictionFromAPL'
[string "@Interface/AddOns/Hekili/Core.lua"]:1397: in function `GetNextPrediction'
[string "@Interface/AddOns/Hekili/Core.lua"]:1683: in function <Interface/AddOns/Hekili/Core.lua:1446>

Snapshots:
1) After popping combustion and casting some random spells:  https://pastebin.com/mvX0GT1K
2)  After combustion/buff coming off CD and doing /reload:  https://pastebin.com/YzpthGGg
3)  After item 2 above, but now popping combustion:  https://pastebin.com/twXQ67Cm

I'm not sure if any of this helps.  Sorry!

**Comment by Hekili** [comment]
Install v10.1.7-1.0.2 and post new snapshots if the issue continues.

**Comment by nhawrot** [comment]
Installed new version, still not functioning.  Here all snapshots:

1) Fresh load into game on Firemage using training dummy:  https://pastebin.com/v2w5ix3Z
2) Step 1, then pressing some random spells:  https://pastebin.com/VYi4c9TZ
3) Step 2, then popping combustion and waiting 8 seconds (I counted that out for when it shows up):  https://pastebin.com/7UnvfVeN

**Comment by nhawrot** [comment]
Here's an example of what's happening:
https://github.com/Hekili/hekili/assets/145692930/b9915349-1992-4fb7-9e43-b78aaebca410

**Comment by nhawrot** [comment]
Here's a better example, showing my brewmaster monk and how it loads automatically, versus my fire mage and then when popping combustion:  https://clipchamp.com/watch/7q6DwUtNYTo

**Comment by Hekili** [comment]
This may be resolved with the same solution as #2810.
